posts/qr-code-conundrum.md

```diff
@@ -0,0 +1,85 @@
+
+For the demo we have setup the app to work in React. React is a decent choice to use as it lets you react to all the scanning results (GET IT … REACT … I’ll leave now). The react states are super useful to control what is happening on screen. When a QR code is found by the decode mechanism we have setup a simple popup (react-skylight - a simple popup tool for react apps) to display the result back to the user.
+
+We have left in the option to change camera in the demo - whilst it is not really a core feature of what we wanted to demonstrate it will hopefully give some idea on how to manipulate cameras on devices with more than one (our full application is mainly used on phones and was important to be able to switch between front and back camera). You will notice in the code that we have added a simple check on the device labels to see if it is label ‘back’ so we can attempt to default the camera to the back one upon initial load. There is some functionality in the libraries to use the default camera on the device but you need to logically set this up with your code (and as you will read below it does come with some fun problems from your favourite fruit related manufacturer).
+
+One thing of note to pass on is that whilst the libraries do work on the major OSs, iOS is a bit more prickly about using its hardware - it will force users to accept the use of the camera in the app - despite it being used in the browser. It doesn’t break anything HOWEVER you will need to account for the fact that some users (let’s be real here, by some users I mean most users) will need to agree to the camera being used - and until they do so the app will not recognise any available cameras on the device. From the app's point of view there will be no registered cameras until the user agrees. You will need some functionality in your app to deal with this. We found react was helpful here to allow the page to “reload” upon state changing.
+
+![Sadness](https://github.com/bradooles/blog-usa/blob/master/images/2021/07/qr-code-conundrum-4.gif)
+>Me after realising that iOS was going to be annoying to deal with
+<br/>
+
+## Get your QR on!
+As you can see it’s fairly easy to start using a QR scanner with your device. The ZXing libraries are quite robust and shockingy responsive (maybe it is just me but I fully expected some crazy delay). From our own experience we have found using the above libraries to work really well in an app context, so much so that I would absolutely use them again to solve a QR scanning problem. In particular it made the implementation of a web application where QR scanning was a main function quite easy to develop.
+
+Below are the various links to references in this blog.
+
+Until next time, happy QR scanning and good luck!
+
+### References
```

Comment:
This probably should be a second level header as well.

Reply:
Updated to second level header now